### PR TITLE
[YulIR] Change expression parts to use strings instead of numbers.

### PR DESCRIPTION
**Pull request by chriseth**
### Checklist
- [ ] Code compiles correctly
- [ ] All tests are passing
- [ ] New tests have been created which fail without the change (if possible)
- [ ] README / documentation was extended, if necessary
- [ ] Changelog entry (if change is visible to the user)
- [ ] Used meaningful commit messages

### CODE REVIEW

**Comment by chriseth** [comment]
This will make it much easier to set gas, value and salt parameters on functions. I think it is also much easier to read and verify.

File: libsolidity/codegen/ir/IRGenerationContext.cpp [hunk 1]
```diff
@@ -108,12 +108,11 @@ string IRGenerationContext::variable(Expression const& _expression)
 		return suffixedVariableNameList(move(var) + "_", 1, 1 + size);
 }
 
-string IRGenerationContext::variablePart(Expression const& _expression, size_t _part)
+string IRGenerationContext::variablePart(Expression const& _expression, std::string const& _part)
```
Comment (erak) [inline libsolidity/codegen/ir/IRGenerationContext.cpp]:
```suggestion
string IRGenerationContext::variablePart(Expression const& _expression, string const& _part)
```

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -1285,7 +1285,7 @@ ostream& IRGeneratorForStatements::defineExpression(Expression const& _expressio
 	return m_code;
 }
 
-ostream& IRGeneratorForStatements::defineExpressionPart(Expression const& _expression, size_t _part)
+ostream& IRGeneratorForStatements::defineExpressionPart(Expression const& _expression, std::string const& _part)
```
Comment (erak) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]:
```suggestion
ostream& IRGeneratorForStatements::defineExpressionPart(Expression const& _expression, string const& _part)
```

**Review by erak — CHANGES_REQUESTED** [review]
Just 2 tiny comments. Looks good otherwise.

**Pull request merged**